### PR TITLE
feat(EMI-1727): create updateCollectionsMutation

### PR DESCRIPTION
**Pull request by fladson**
Port the work done at https://github.com/artsy/gravity/pull/17431 to metaphysics.

Example request for reference:
```
  mutation {
    updateMeCollectionsMutation(
      input: {
        attributes: [
          {
            id: ""
            shareableWithPartners: false
          },
          {
            id: ""
            shareableWithPartners: true
          },
        ]
      }
    ) {
      meCollectionsOrErrors {
        ... on UpdateMeCollectionsSuccess {
          collection {
            id
          	shareableWithPartners
            name
            artworksCount
          }
        }
        ... on UpdateMeCollectionsFailure {
          mutationError {
            type
            message
          }
        }
      }
    }
  }
```

Successful request
SCR-20240325-owt

Failed request
SCR-20240325-ox2

@artsy/emerald-devs

### CODE REVIEW

File: _schemaV2.graphql [hunk 1]
```diff
@@ -13423,6 +13428,11 @@ type Mutation {
   # Update a collection
   updateCollection(input: updateCollectionInput!): updateCollectionPayload
 
+  # Updates the user's collections in batch.
+  updateCollectionsMutation(
```
Comment (starsirius) [inline _schemaV2.graphql:13432]: Naming suggestion: Given the mutation names are global in the schema, without name spacing, let's prepare ourselves for 2 different mutations:

- update collections for myself (based on the access token): our needs
- update collections for a user (e.g. performed by an admin, with a user id param): future use cases

I can't find a consistent pattern in MP for mutations, but I'd suggest we use name

- `updateMeCollectionsMutation`: update collections for myself (I kind of like `updateMyCollectionsMutation` but apparently "MyCollection" means something else already 😞 .)
- `updateUserCollectionsMutation`: update collections for a user

Comment (starsirius) [reply _schemaV2.graphql:13432]: And we can also update the input, payload, response types accordingly.

File: _schemaV2.graphql [hunk 2]
```diff
@@ -4108,6 +4108,11 @@ type CollectionsEdge {
   node: Collection
 }
 
+input CollectionsInput {
```
Comment (starsirius) [inline _schemaV2.graphql:4111]: Naming suggestion: The plural collection"s" here sounds a bit unintuitive with the singular data inside. Maybe we can call it `UpdateMeCollectionInput` so the mutation below reads a bit nicer:

```graphql
input updateMeCollectionsMutationInput {
  attributes: [UpdateMeCollectionInput!]!
  clientMutationId: String
}
```

(There is an existing [`updateCollectionInput`](https://github.com/artsy/metaphysics/blob/0f0cf799cd780744a9e2ce8ac87d7ca01c832c65/_schemaV2.graphql#L18868-L18874) that requires an id and a name. 🤷  The collection modeling for multiple purposes is kind of making naming tricky.)

File: src/schema/v2/me/updateCollectionsMutation.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    })
+
+    try {
+      return await meUpdateCollectionsLoader({
+        attributes: JSON.stringify(gravityPayload),
+      })
+    } catch (error) {
+      const formattedErr = formatGravityError(error)
+      if (formattedErr) {
+        return [{ ...formattedErr, _type: "GravityMutationError" }]
```
Comment (starsirius) [inline src/schema/v2/me/updateCollectionsMutation.ts:99]: The corresponding Gravity batch endpoint returns

- An array of updated collections when succeeded
- A single [_object_](https://github.com/artsy/gravity/blob/fa66526dafd211963f9db0fe677e4518c4e610d4/spec/api/v1/me_collections_endpoint_spec.rb#L115-L116) of an error when failed.

Here in MP we turn the error response type into an array as well. I guess it's fine (or even more flexible in the future a Gravity batch endpoint can return multiple errors).

**Review by starsirius — COMMENTED** [review]
LGTM. I have a few naming suggestions otherwise looking good! 👍

File: _schemaV2.graphql (hunk 1 above)
Comment (fladson) [reply _schemaV2.graphql]: 👍

**Pull request merged**